.github/workflows/build-and-test-snap.yml

```diff
@@ -41,9 +84,22 @@ jobs:
           name: snap-logs
           path: tests/logs/*.log
 
-      - name: Upload artifact
-        uses: actions/upload-artifact@v4
-        with:
-          name: matter-pi-gpio-commander_${{ github.run_number}}_arm64.snap
-          path: ${{ steps.snapcraft.outputs.snap }}
-  
\ No newline at end of file
+  promote-beta-arm64:
+    # Only promote if we are on the main branch
+    if: github.ref == 'refs/heads/main'
+    runs-on: ubuntu-latest
+    needs: [build-arm64, test-arm64]
+    steps:
+      - name: Install Snapcraft
+        # Install snapcraft and add it to PATH
+        run: |
+          sudo snap install snapcraft --classic
+          echo /snap/bin >> $GITHUB_PATH
+        shell: bash
+
+      - name: Promote Snap
+        env: # Workaround for https://github.com/snapcore/snapcraft/issues/4439
+          SNAPCRAFT_HAS_TTY: "true"
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.STORE_LOGIN }}
+        run: |
+          yes | snapcraft promote matter-pi-gpio-commander --from-channel latest/edge --to-channel latest/beta
```

Comment:
There could be a race condition where `latest/edge` got updated after this workflow published, and before it can promote - making it promote a different snap revision than the one we tested.

1. Can we publish the same artifact to `latest/beta`?
1. Can we get the revision number from the publish step, and use that in the promote?

Reply:
Approach 1 fails:

```
Publishing snap "matter-pi-gpio-commander_2.0+chip-5bff3a5a_arm64.snap"...
Installing Snapcraft...
/usr/bin/sudo snap install --classic snapcraft
snapcraft 8.3.2 from Canonical** installed
/snap/bin/snapcraft upload matter-pi-gpio-commander_2.0+chip-5bff3a5a_arm64.snap --release latest/beta/test
Uploading... (--->)
Uploading... (<---)
Status: processing
Status: error while processing
Issues while processing snap:
- binary_sha3_384: Error checking upload uniqueness.
Full execution log: '/home/runner/.local/state/snapcraft/log/snapcraft-20240828-111452.881[9](https://github.com/canonical/matter-pi-gpio-commander/actions/runs/10595393598/job/29361633789#step:3:10)83.log'
Error: The process '/snap/bin/snapcraft' failed with exit code 1
```

Reply:
Moved this to an issue https://github.com/canonical/matter-pi-gpio-commander/issues/83